5.machine_learning/scripts/3.extract_model_coef.py

```diff
@@ -156,3 +156,22 @@
 # Show the top ten ranking features for predicting "Failing" class
 coefficients_failing_df.head(10)
 
+
+# ## Add ranking column with sorted descending values and save the CSV for visualization
+
+# In[10]:
+
+
+# Sort coefficients_df by descending order
+coefficients_df = coefficients_df.sort_values(by='Coefficient', ascending=False)
+
+# Add a new column 'Rank'
+coefficients_df['Rank'] = range(1, len(coefficients_df) + 1)
+
+# Save the ranked df
+coefficients_df.to_csv(f'{data_dir}/ranked_coeffs.csv', index=False)
+
+# Show df to assess if the ranking was performed correctly
```

Comment:
Would it make sense to add a programmatic check of some kind to validate the shape of the dataframe (raising an error if there's some sort of misalignment)?

Reply:
In this case, I think that it is okay not to include a programmatic check here since the main thing to see is that the rank starts at the most positive value and the size is correct. But, I will keep this in mind in the future since have checks is important!